SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -1133,3 +1133,212 @@ index 0558611..18c84fc 100644
 -- 
 2.34.1
 
+
+From 729e3374fbd2dd54c2124eb8e16664cddf5ac2ba Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Wed, 28 Aug 2024 16:26:12 -0700
+Subject: [PATCH 2/2] Add snapshotexclude feature to remove a specific repo
+ from the snapshot
+
+---
+ client/prototypes.h                        |  6 +++
+ client/repo.c                              |  2 +-
+ client/repolist.c                          | 54 ++++++++++++++++++++++
+ etc/bash_completion.d/tdnf-completion.bash |  7 ++-
+ include/tdnftypes.h                        |  1 +
+ tools/cli/lib/help.c                       |  3 +-
+ tools/cli/lib/parseargs.c                  |  3 +-
+ 7 files changed, 72 insertions(+), 4 deletions(-)
+
+diff --git a/client/prototypes.h b/client/prototypes.h
+index cb8a8d3..ddac483 100644
+--- a/client/prototypes.h
++++ b/client/prototypes.h
+@@ -771,6 +771,12 @@ TDNFAlterRepoState(
+     const char* pszId
+     );
+ 
++uint32_t
++TDNFExcludeFromSnapshot(
++    PTDNF_REPO_DATA pRepos,
++    const char* pszId
++    );
++
+ uint32_t
+ TDNFCloneRepo(
+     PTDNF_REPO_DATA pRepoIn,
+diff --git a/client/repo.c b/client/repo.c
+index 0331796..c675821 100644
+--- a/client/repo.c
++++ b/client/repo.c
+@@ -99,7 +99,7 @@ TDNFInitRepo(
+     pRepo->appdata = pSolvRepoInfo;
+ 
+     if (pRepoData->nHasMetaData) {
+-        if (pszSnapshotTime != NULL) {
++        if (!pRepoData->nExcludeSnapshot && pszSnapshotTime != NULL) {
+             dwError = TDNFInitRepoFromMetadata(pRepo, pRepoData->pszId, pRepoMD, pszSnapshotTime);
+             BAIL_ON_TDNF_ERROR(dwError);
+         } else {
+diff --git a/client/repolist.c b/client/repolist.c
+index df37959..40146de 100644
+--- a/client/repolist.c
++++ b/client/repolist.c
+@@ -358,6 +358,7 @@ TDNFCreateRepo(
+     BAIL_ON_TDNF_ERROR(dwError);
+ 
+     pRepo->nEnabled = TDNF_REPO_DEFAULT_ENABLED;
++    pRepo->nExcludeSnapshot = 0;
+     pRepo->nHasMetaData = 1;
+     pRepo->nSkipIfUnavailable = TDNF_REPO_DEFAULT_SKIP;
+     pRepo->nGPGCheck = TDNF_REPO_DEFAULT_GPGCHECK;
+@@ -695,6 +696,12 @@ TDNFRepoListFinalize(
+                           1,
+                           pSetOpt->pszOptValue);
+         }
++        else if (strcmp(pSetOpt->pszOptName, "excludesnapshot") == 0)
++        {
++            dwError = TDNFExcludeFromSnapshot(
++                pTdnf->pRepos,
++                pSetOpt->pszOptValue);
++        }
+         BAIL_ON_TDNF_ERROR(dwError);
+     }
+ 
+@@ -787,6 +794,53 @@ error:
+     goto cleanup;
+ }
+ 
++uint32_t
++TDNFExcludeFromSnapshot(
++    PTDNF_REPO_DATA pRepos,
++    const char* pszId
++    )
++{
++    uint32_t dwError = 0;
++    int nIsGlob = 0;
++    if(!pRepos && IsNullOrEmptyString(pszId))
```

Comment:
&& or || here?

Reply:
Good catch, this is actually copy paste from upstream, we can likely contribute this small fix upstream if not already present